tests/bwc/test_upgrade.py

```diff
@@ -649,3 +649,23 @@ def test_insert_returning_in_mixed_cluster(self):
                                                               'Returning clause for Insert is only supported when all '
                                                               'nodes in the cluster running at least version 4.2.0\\]'):
                     cursor.execute('''INSERT INTO test VALUES(2, 'msg') returning id''')
+
+    def test_insert_in_mixed_cluster(self):
+
+        """Test backward compatibility when using a plain insert in a mixed cluster
+           of 4.1 and 4.2 nodes. Plain inserts without return values or update on conflict
+           became an own operation for efficienty reasons. In case of a mixed cluster
+           the system falls back to use the generic upsert.
+        """
+
+        with connect(self.node_4_1.http_url, error_trace=True) as conn_4_1:
+            cursor = conn_4_1.cursor()
+            cursor.execute('CREATE TABLE test (id int primary key, message string) clustered into 2 shards;')
+            self.assertEqual(cursor.rowcount, 1)
+            cursor.execute('''INSERT INTO test VALUES(1, 'fallback to upsert')''')
```

Comment:
Could you do some more inserts here with different ids to make sure the records are sent to all nodes in the cluster? Could be that `1` will remain local on the 4.1 node.